### DISCUSSION TITLE
H2 Display Issue

### DISCUSSION THREAD

**Discussion by gsherwin74** [kachurovskiy/nanoels#95] [General]
Just finished assembling the project and uploading the code.  First time turning it on doing some bench testing, when I spin the encoder, the position and rpm lines on the display disappear and reappear when the encoder stops.  The display still shows the lines above those bottom two no matter the encoder movement.

When the encoder stops rotating, the bottom two rows of the screen appear again and it shows an updated position.  So it knows its moving but you can't see the rpm and distance moving as it's happening.

Any ideas?

**Comment by kachurovskiy** [comment]
Hi @gsherwin74, this is intentional: updating display takes 40ms and we can't issue step commands during this time. That means the screen can't be updated when the stepper is running. At the same time it'd be undesirable to show no longer accurate position information so we hide it before we start moving the stepper.

https://github.com/kachurovskiy/nanoels/blob/main/h2/h2.ino#L283

**Reply by gsherwin74** [reply]
Will RPM be shown when the ELS is off?

**Reply by kachurovskiy** [reply]
Yes.
